scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -506,7 +506,11 @@ public final void processMessage(Record record) throws HandshakeException {
 						}
 					}
 					expectMessage(messageToProcess);
-					doProcessMessage(messageToProcess);
+					if (messageToProcess.getContentType() == ContentType.CHANGE_CIPHER_SPEC) {
+						setCurrentReadState();
```

Comment:
I don't know if this is an issue but I think we remove the debug log about receiving a CHANGE_CIPHER_SPEC